pkg/operator/ceph/cluster/cluster.go

```diff
@@ -373,6 +375,116 @@ func (c *ClusterController) preClusterStartValidation(cluster *cluster, clusterO
 	return nil
 }
 
+func extractExitCode(err error) (int, bool) {
+	exitErr, ok := err.(*exec.ExitError)
```

Comment:
This seems to be causing an error in the integration tests, for example [here](https://jenkins.rook.io/blue/rest/organizations/jenkins/pipelines/rook/pipelines/rook/branches/PR-5483/runs/29/nodes/61/steps/131/log/?start=0). Did this change in your latest push? The CI was passing before.
```
    suite.go:62: test panicked: interface conversion: error is *exec.Error, not *exec.ExitError
```

Reply:
Nope, didn’t touch that piece. Must’ve been some update, e.g. through the rebase.

The only traceback related to ExitError I can find in the run you linked does not relate to my code either:

```
--- FAIL: TestCephHelmSuite (19.58s)
    suite.go:62: test panicked: interface conversion: error is *exec.Error, not *exec.ExitError
        goroutine 653 [running]:
        runtime/debug.Stack(0xc0008ab528, 0x184a640, 0xc000826960)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9d
        github.com/stretchr/testify/suite.failOnPanic(0xc000836800)
        	/home/rook/go/pkg/mod/github.com/stretchr/testify@v1.4.0/suite/suite.go:62 +0x57
        panic(0x184a640, 0xc000826960)
        	/usr/local/go/src/runtime/panic.go:679 +0x1b2
        github.com/rook/rook/pkg/util/exec.runCommandWithOutput(0xc0009278c0, 0x0, 0xc0009437c0, 0x2, 0x2, 0xc0009278c0)
        	/home/rook/go/src/github.com/rook/rook/pkg/util/exec/exec.go:285 +0x2b2
        github.com/rook/rook/pkg/util/exec.(*CommandExecutor).ExecuteCommandWithOutput(0x2b38218, 0x1a5864b, 0x4, 0xc0009437c0, 0x2, 0x2, 0x10, 0x0, 0x18, 0x20)
        	/home/rook/go/src/github.com/rook/rook/pkg/util/exec/exec.go:128 +0xa0
        github.com/rook/rook/tests/framework/utils.(*HelmHelper).Execute(0xc00091ede0, 0xc0009437c0, 0x2, 0x2, 0x10, 0x0, 0x0, 0x0)
        	/home/rook/go/src/github.com/rook/rook/tests/framework/utils/helm_helper.go:42 +0x7c
        github.com/rook/rook/tests/framework/utils.(*HelmHelper).GetLocalRookHelmChartVersion(0xc00091ede0, 0x1a66bfc, 0xf, 0x1, 0x8, 0xc000583878, 0x10fd01e)
        	/home/rook/go/src/github.com/rook/rook/tests/framework/utils/helm_helper.go:55 +0xa6
        github.com/rook/rook/tests/framework/installer.(*CephInstaller).CreateRookOperatorViaHelm(0xc00074f0e0, 0x1a5bd64, 0x7, 0x1a7a146, 0x1b, 0x1, 0x1)
        	/home/rook/go/src/github.com/rook/rook/tests/framework/installer/ceph_installer.go:179 +0x6e
        github.com/rook/rook/tests/framework/installer.(*CephInstaller).InstallRook(0xc00074f0e0, 0x1a5bd64, 0x7, 0x1a5e602, 0x9, 0x2860d00, 0x1a57648, 0x0, 0x1, 0xc000583b01, ...)
        	/home/rook/go/src/github.com/rook/rook/tests/framework/installer/ceph_installer.go:462 +0x129
        github.com/rook/rook/tests/integration.(*TestCluster).Setup(0xc0002dde00)
        	/home/rook/go/src/github.com/rook/rook/tests/integration/ceph_base_deploy_test.go:169 +0xd3
        github.com/rook/rook/tests/integration.StartTestCluster(0xc00084d930, 0xc0002dde00, 0x1a51820, 0x409379)
        	/home/rook/go/src/github.com/rook/rook/tests/integration/ceph_base_deploy_test.go:161 +0x303
        github.com/rook/rook/tests/integration.(*HelmSuite).SetupSuite(0xc0008a00c0)
        	/home/rook/go/src/github.com/rook/rook/tests/integration/ceph_helm_test.go:80 +0x15f
        github.com/stretchr/testify/suite.Run(0xc000836800, 0x1d66500, 0xc0008a00c0)
        	/home/rook/go/pkg/mod/github.com/stretchr/testify@v1.4.0/suite/suite.go:102 +0x4ff
        github.com/rook/rook/tests/integration.TestCephHelmSuite(0xc000836800)
        	/home/rook/go/src/github.com/rook/rook/tests/integration/ceph_helm_test.go:52 +0xae
        testing.tRunner(0xc000836800, 0x1b68a20)
        	/usr/local/go/src/testing/testing.go:909 +0xc9
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x350
```

Looking at the docs of [Error](https://golang.org/pkg/os/exec/#Error), it seems to me as something in the environment is borked, maybe the `ceph` executable is missing?

(Given that I use `foo, ok := bar.(*type)` and not `foo := bar.(*type)`, I would also expect `ok` to simply be false instead of a panic.)

Reply:
@travisn So this is not related to my changes? Should I just wait for a fix and rebase afterward?

Reply:
Yes, we will plan on getting that fixed in the next couple days, then you can rebase, thanks.

Reply:
Fixed here https://github.com/rook/rook/pull/6503

Reply:
@horazont #6503 is merged now, so you can go ahead and rebase to pick it up